convert2rhel/grub.py

```diff
@@ -0,0 +1,597 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import re
+import shutil
+
+from convert2rhel import systeminfo, utils
+
+logger = logging.getLogger(__name__)
+
+EFI_MOUNTPOINT = "/boot/efi/"
+"""The path to the required mountpoint for ESP."""
+
+CENTOS_EFIDIR_CANONICAL_PATH = os.path.join(EFI_MOUNTPOINT, "EFI/centos/")
+"""The canonical path to the default efi directory on for CentOS Linux system."""
+
+RHEL_EFIDIR_CANONICAL_PATH = os.path.join(EFI_MOUNTPOINT, "EFI/redhat/")
+"""The canonical path to the default efi directory on for RHEL system."""
+
+# TODO(pstodulk): following constants are valid only for x86_64 arch
+DEFAULT_INSTALLED_EFIBIN_FILENAMES = ("shimx64.efi", "grubx64.efi")
+"""Filenames of the EFI binary files that could be by default instaled on the system.
+
+Sorted by the recommended preferences. The first one is most preferred, but it
+is provided by the shim rpm which does not have to be installed on the system.
+In case it's missing, another files should be used instead.
```

Comment:
Discussed in person, but addint the comment for other possible readers :)

In case of the secure boot the `shimx64` is required, otherwise any of these files can be used. But in case a user will want to switch to the secure boot (not during the conversion but whenever in future), it's easier for them to do it when the `shimx64` file is used. So the `shimx64` is prefered to use because of this reason only.